lib/builders/windows-builder.ts

```diff
@@ -0,0 +1,231 @@
+                    role: blueprints.getResource(context => {
+                        return new iam.Role(context.scope, 'ClusterRole', { 
+                            assumedBy: new iam.ServicePrincipal("eks.amazonaws.com"),
+                            managedPolicies: [
+                                iam.ManagedPolicy.fromAwsManagedPolicyName("AmazonEKSClusterPolicy"),
+                                iam.ManagedPolicy.fromAwsManagedPolicyName("AmazonEKSVPCResourceController")
+                            ]
+                        });
+                    }),
+                    managedNodeGroups: [
+                        addGenericNodeGroup(options),
+                        addWindowsNodeGroup(options)
+                    ]
+                })
+            )
+            .addOns(
+                new blueprints.NestedStackAddOn({
+                    id: "usage-tracking-addon",
+                    builder: UsageTrackingAddOn.builder(),
+                })
+            );
+        return builder;
+    }
+}
+
+/**
+ * Nested stack that is used as tracker for Windows Accelerator
+ */
+export class UsageTrackingAddOn extends NestedStack {
+
+    static readonly USAGE_ID = "qs-1ubotj5kl";
+
+    public static builder(): blueprints.NestedStackBuilder {
+        return {
+            build(scope: Construct, id: string, props: NestedStackProps) {
+                return new UsageTrackingAddOn(scope, id, props);
+            }
+        };
+    }
+
+    constructor(scope: Construct, id: string, props: NestedStackProps) {
+        super(scope, id, utils.withUsageTracking(UsageTrackingAddOn.USAGE_ID, props));
+    }
+}
+
+/**  
+ * This function adds a generic node group to the windows cluster.
+ * @param: options: WindowsOptions
+ * @returns: blueprints.ManagedNodeGroup
+ */
+function addGenericNodeGroup(options: WindowsOptions): blueprints.ManagedNodeGroup {
+
+    return {
+        id: "mng-linux",
+        amiType: NodegroupAmiType.AL2_X86_64,
+        instanceTypes: [new ec2.InstanceType('m5.4xlarge')],
+        desiredSize: options.desiredNodeSize ?? defaultNodeProps.desiredNodeSize, 
+        minSize: options.minNodeSize ?? defaultNodeProps.minNodeSize, 
+        maxSize: options.maxNodeSize ?? defaultNodeProps.maxNodeSize,
+        nodeRole: blueprints.getNamedResource("node-role") as iam.Role,
+        nodeGroupSubnets: { subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS },
+        launchTemplate: {
+            tags: options.genericNodeGroupTags ?? defaultNodeProps.genericNodeGroupTags,
+            requireImdsv2: false
+        }
+    };
+}
+
+/**
+ * This function adds a windows node group to the windows cluster.
+ * @param options: WindowsOptions
+ * @returns: blueprints.ManagedNodeGroup
+ */
+function addWindowsNodeGroup(options: WindowsOptions): blueprints.ManagedNodeGroup {
+    const result : blueprints.ManagedNodeGroup = {
+        id: "mng-windows",
+        amiType: options.windowsAmiType ?? defaultNodeProps.windowsAmiType,
```

Comment:
recommendation (can be deferred): you don't need to check `options` value with `??` operator in each line. 
We generally, just use `const mergedOptions = merge(options, defaultOptions)` and use mergedOptions without extra checks.

Reply:
Fixed.